lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -523,6 +523,34 @@ def getvm(self, name=None, uuid=None, folder=None):
 
         return vm
 
+    def get_datacenter(self):
```

Comment:
I am trying to understand - why this refactoring is required.

Reply:
Because the new function `get_folder_path` will be called twice, and it needs to get the path of datacenter via `Datacenter` object.